### PR TITLE
Fixes #9708: Create a relay API for shared-files

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/9708

Initialize with './virtualenv.py flask && . flask/bin/activate && pip install -r requirements.txt'
test with ./run.py

### CODE REVIEW

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

File: rudder-server-relay/SOURCES/relay-api/relay_api/views.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    nodes = get_nodes_list(NODESLIST_FILE)
+    file_hash = request.args.get('hash', '')
+    if target_uuid not in nodes:
+      # forward the request if the node is unknown
+      #policy_server = get_file_content(POLICY_SERVER_FILE)
+      policy_server = "localhost"
+      url = "https://"+policy_server+"/rudder/relay-api/shared-files/" + target_uuid + "/" + source_uuid + "/" + file_id + "?hash=" + file_hash
+      res = shared_files_head_forward(url)
+    else:
+      #my_uuid = get_my_uuid(UUID_FILE)
```
Comment (amousset) [inline rudder-server-relay/SOURCES/relay-api/relay_api/views.py]: This looks like debug code.

**Comment by amousset** [comment]
```json
{
    "c5e38f75-3fbe-402f-ac78-4ada2ea784a5": {
        "hostname": "host1.rudder.local",
        "key-hash": "sha1:da39a3ee5e6b4b0d3255bfef95601890afd80709",
        "policy-server": "9b25d384-a0ed-4ac1-adf2-b5b0277a455d"
    },
    "6b26bf62-fe19-45ba-a0bc-4a1ddeca52ef": {
        "hostname": "host2.rudder.local",
        "key-hash": "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
        "policy-server": "9b25d384-a0ed-4ac1-adf2-b5b0277a455d"
    },
    "465b0ed4-6cad-4dc3-a0ec-cf84c8e1d752": {
        "hostname": "host3.rudder.local",
        "key-hash": "cf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e",
        "policy-server": "6b26bf62-fe19-45ba-a0bc-4a1ddeca52ef"
    },
    "bf745c11-e995-4b4c-806a-86604708cfc3": {
        "hostname": "host4.rudder.local",
        "key-hash": "sha1:0539a3ee5e6b4b0d3255bfef95601890afd80710",
        "policy-server": "465b0ed4-6cad-4dc3-a0ec-cf84c8e1d752"
    }
}
```

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

File: rudder-server-relay/SOURCES/relay-api/relay_api/shared_files.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+# ======================
+#  Forward PUT API call
+# ======================
+# Parameters:
+# - stream     stream of posted data
+# - url        where to forward to
+# Returns the full path of the created file
+def shared_files_put_forward(stream, url):
+  # This is needed to make requests now the length
```
Comment (amousset) [inline rudder-server-relay/SOURCES/relay-api/relay_api/shared_files.py]: know

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**